src/rest.cpp

```diff
@@ -487,26 +487,33 @@ static bool rest_getutxos(HTTPRequest* req, const std::string& strURIPart)
     std::vector<bool> hits;
     bitmap.resize((vOutPoints.size() + 7) / 8);
     {
-        LOCK2(cs_main, mempool.cs);
-
-        CCoinsView viewDummy;
-        CCoinsViewCache view(&viewDummy);
-
-        CCoinsViewCache& viewChain = *pcoinsTip;
-        CCoinsViewMemPool viewMempool(&viewChain, mempool);
-
-        if (fCheckMemPool)
-            view.SetBackend(viewMempool); // switch cache backend to db+mempool in case user likes to query mempool
-
-        for (size_t i = 0; i < vOutPoints.size(); i++) {
-            bool hit = false;
-            Coin coin;
-            if (view.GetCoin(vOutPoints[i], coin) && !mempool.isSpent(vOutPoints[i])) {
-                hit = true;
-                outs.emplace_back(std::move(coin));
+        auto process_utxos = [&vOutPoints, &outs, &hits](const CCoinsViewCache& view, const CTxMemPool& mempool) {
+            for (const COutPoint& vOutPoint : vOutPoints) {
+                bool hit = false;
+                Coin coin;
+                if (view.GetCoin(vOutPoint, coin) && !mempool.isSpent(vOutPoint)) {
+                    hit = true;
+                    outs.emplace_back(std::move(coin));
+                }
+                hits.push_back(hit);
             }
+        };
+
+        if (fCheckMemPool) {
+            // use db+mempool as cache backend in case user likes to query mempool
+            LOCK2(cs_main, mempool.cs);
+            CCoinsViewCache& viewChain = *pcoinsTip;
+            CCoinsViewMemPool viewMempool(&viewChain, mempool);
+            CCoinsViewCache view(&viewMempool);
```

Comment:
There's no need for an intermediary cache here anymore. You can make the lambda take a `CCoinsView&` instead, and either pass it `viewMempool` (here) or `*pcoinsTip` (below) directly.

Reply:
You're right, thanks for the catch!
Will amend the commit, and update this PR.